biggus/__init__.py

```diff
@@ -2781,71 +2826,162 @@ def ndarray(self):
+absolute = _ufunc_wrapper(np.absolute)
+rint = _ufunc_wrapper(np.rint)
+sign = _ufunc_wrapper(np.sign)
+conj = _ufunc_wrapper(np.conj)
+exp = _ufunc_wrapper(np.exp)
+exp2 = _ufunc_wrapper(np.exp2)
+log = _ufunc_wrapper(np.log)
+log2 = _ufunc_wrapper(np.log2)
+log10 = _ufunc_wrapper(np.log10)
+expm1 = _ufunc_wrapper(np.expm1)
+sqrt = _ufunc_wrapper(np.sqrt)
+square = _ufunc_wrapper(np.square)
+reciprocal = _ufunc_wrapper(np.reciprocal)
+
+
+# Dual argument math operations.
+add = _ufunc_wrapper(np.add)
+sub = _ufunc_wrapper(np.subtract)
+subtract = _ufunc_wrapper(np.subtract)
+multiply = _ufunc_wrapper(np.multiply)
+floor_divide = _ufunc_wrapper(np.floor_divide)
+true_divide = _ufunc_wrapper(np.true_divide)
+divide = _ufunc_wrapper(np.divide)
+power = _ufunc_wrapper(np.power)
+
+
+# Single argument trigonometric functions.
+sin = _ufunc_wrapper(np.sin)
+cos = _ufunc_wrapper(np.cos)
+tan = _ufunc_wrapper(np.tan)
+arcsin = _ufunc_wrapper(np.arcsin)
+arccos = _ufunc_wrapper(np.arccos)
+arctan = _ufunc_wrapper(np.arctan)
+sinh = _ufunc_wrapper(np.sinh)
+cosh = _ufunc_wrapper(np.cosh)
+tanh = _ufunc_wrapper(np.tanh)
+arcsinh = _ufunc_wrapper(np.arcsinh)
+arccosh = _ufunc_wrapper(np.arccosh)
+arctanh = _ufunc_wrapper(np.arctanh)
+deg2rad = _ufunc_wrapper(np.deg2rad)
+rad2deg = _ufunc_wrapper(np.rad2deg)
+
+
+# Dual argument trigonometric functions.
+arctan2 = _ufunc_wrapper(np.arctan2)
+hypot = _ufunc_wrapper(np.hypot)
+
+
+# Bit-twiddling functions.
+bitwise_and = _ufunc_wrapper(np.bitwise_and)
+bitwise_or = _ufunc_wrapper(np.bitwise_or)
+bitwise_xor = _ufunc_wrapper(np.bitwise_xor)
+invert = _ufunc_wrapper(np.invert)
+left_shift = _ufunc_wrapper(np.left_shift)
+right_shift = _ufunc_wrapper(np.right_shift)
+
+
+# Comparisson functions.
+greater = _ufunc_wrapper(np.greater)
+greater_equal = _ufunc_wrapper(np.greater_equal)
+less = _ufunc_wrapper(np.less)
+less_equal = _ufunc_wrapper(np.less_equal)
+not_equal = _ufunc_wrapper(np.not_equal)
+equal = _ufunc_wrapper(np.equal)
+logical_and = _ufunc_wrapper(np.logical_and)
+logical_or = _ufunc_wrapper(np.logical_or)
+logical_xor = _ufunc_wrapper(np.logical_xor)
+logical_not = _ufunc_wrapper(np.logical_not)
+maximum = _ufunc_wrapper(np.maximum)
+minimum = _ufunc_wrapper(np.minimum)
+fmax = _ufunc_wrapper(np.fmax)
+fmin = _ufunc_wrapper(np.fmin)
+
+
+# Floating functions.
+isreal = _unary_fn_wrapper('np.isreal', np.isreal)
+iscomplex = _unary_fn_wrapper('np.iscomplex', np.iscomplex)
+isinf = _unary_fn_wrapper('np.isinf', np.isinf)
+isnan = _unary_fn_wrapper('np.isnan', np.isnan)
+signbit = _unary_fn_wrapper('np.signbit', np.signbit)
+copysign = _dual_input_fn_wrapper('np.copysign', np.copysign)
+nextafter = _dual_input_fn_wrapper('np.nextafter', np.nextafter)
+# modf = _unary_fn_wrapper('np.modf', np.modf)  # Needs 2 arrays out.
+ldexp = _dual_input_fn_wrapper('np.ldexp', np.ldexp)
+# frexp = _unary_fn_wrapper('np.frexp', np.frexp)  # Needs 2 arrays out.
+fmod = _dual_input_fn_wrapper('np.fmod', np.fmod)
```

Comment:
Why does this set of functions call the specific wrappers directly rather than making use of the general `ufunc_wrapper`?